File: frontend/src/types/sticker.ts
export interface ISticker {
	id: string;
	pack_id: number;
	pack_name?: string;
	name: string;
	image_url: string;
	sort_order?: number;
	is_active: boolean;
	created_at?: string;
	updated_at?: string;
}

export interface IStickerPack {
	id: number;
	name: string;
	description?: string | null;
	thumbnail_url?: string | null;
	is_active: boolean;
	created_at?: string;
	updated_at?: string;
}

File: messenger-service/internal/infrastructure/persistence/model/sticker.go
package model

import "time"

type StickerPack struct {
	ID           uint    `gorm:"primaryKey"`
	Name         string  `gorm:"uniqueIndex;size:255;not null"`
	Description  *string `gorm:"type:text"`
	ThumbnailURL *string `gorm:"size:500"`
	IsActive     bool    `gorm:"default:true"`
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

func (StickerPack) TableName() string { return "sticker_packs" }

type Sticker struct {
	ID        uint   `gorm:"primaryKey"`
	PackID    uint   `gorm:"index;not null"`
	Name      string `gorm:"size:255;not null"`
	ImageURL  string `gorm:"size:500;not null"`
	SortOrder int    `gorm:"default:0"`
	IsActive  bool   `gorm:"default:true"`
	CreatedAt time.Time
	UpdatedAt time.Time
}

func (Sticker) TableName() string { return "stickers" }
